types/topology.go

```diff
@@ -691,77 +872,38 @@ func (t *Topology) ImportEnvs() {
 	}
 }
 
-func (t *Topology) GetNodeDns(name string) *DNSConfig {
-	if ndef, ok := t.Nodes[name]; ok {
-		nodeDNS := ndef.GetDns()
-		if nodeDNS != nil {
-			return nodeDNS
-		}
-
-		groupDNS := t.GetGroup(t.GetNodeGroup(name)).GetDns()
-		if groupDNS != nil {
-			return groupDNS
-		}
-
-		kindDNS := t.GetKind(t.GetNodeKind(name)).GetDns()
-		if kindDNS != nil {
-			return kindDNS
-		}
-	}
-
-	defaultDNS := t.GetDefaults().GetDns()
-	if defaultDNS == nil {
-		// initialize defaultDNS to an empty DNSConfig struct
-		// so that we don't have to check for nil in the caller
-		defaultDNS = &DNSConfig{}
-	}
-
-	return defaultDNS
-}
-
 // GetCertificateConfig returns the certificate configuration for the given node.
-func (t *Topology) GetCertificateConfig(name string) *CertificateConfig {
+func (t *Topology) GetCertificateConfig(nodeName string) *CertificateConfig {
```

Comment:
can this benefit from `getField`?

Reply:
this one is/was because the merging. I think we could getField if we just wanted the first set value but this one merges them all how the string slice/map ones do and since it was the only one of this flavor I left it alone